Gcpe.Hub.API/Controllers/MessagesController.cs

```diff
@@ -133,38 +125,27 @@ public IActionResult UpdateMessage(Guid id, Models.Message message)
                 var dbMessage = dbContext.Message.Find(id);
                 if (dbMessage != null && dbMessage.IsActive)
                 {
-                    if (!dbMessage.IsPublished && message.IsPublished)
+                    bool wasPublished = dbMessage.IsPublished;
+                    if (wasPublished && message.IsPublished)
                     {
-                        message.SortOrder = 0;
-                        BumpSortOrders(1, 0, null);
-                    }
-                    else
-                    {
-                        // going up!
-                        if (dbMessage.SortOrder > message.SortOrder)
-                        {
-                            BumpSortOrders(1, message.SortOrder, dbMessage.SortOrder);
-                        }
-                        // going down¡
-                        else if (dbMessage.SortOrder < message.SortOrder)
+                        if (message.SortOrder != dbMessage.SortOrder)
                         {
-                            BumpSortOrders(-1, dbMessage.SortOrder, message.SortOrder);
+                            // -1 going up, 0 going down!
+                            bool up = message.SortOrder < dbMessage.SortOrder;
+                            IQueryable<Message> messages = dbContext.Message.Where(m => m.IsPublished && m.IsActive);
+                            Message messageToSwap = (up ? messages.Where(m => m.SortOrder < dbMessage.SortOrder).OrderByDescending(m => m.SortOrder)

```

Comment:
what happens if an API request is made with a sort order that isn't a simple move up or down (eg old sort order of 10 and update request with sort order 0)